internal/host/hostcommands/install_cmd.go

```diff
@@ -73,7 +73,11 @@ func (i *installCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models
 		return nil, err
 	}
 
-	infraEnv, err := common.GetInfraEnvByClusterFromDB(i.db, *cluster.ID)
+	if host.InfraEnvID == "" {
+		return nil, errors.New("Hosts without infra envs can no longer be installed")
+	}
```

Comment:
What do you think about loading the infraEnv only when `host.InfraEnvID` is set. This would help us preserve backwards compatibility, at least for the 2.4 time frame.

This would require checking if `infraEnv` is `nil` here: https://github.com/openshift/assisted-service/blob/4cb3d5365d90873eea60d5f3094167ab4914e132/internal/host/hostcommands/install_cmd.go#L291